src/lib/Siteaccess/AbstractSiteaccessPreviewVoter.php

```diff
@@ -43,7 +46,15 @@ public function vote(SiteaccessPreviewVoterContext $context): bool
             null,
             $siteaccess
         );
-        if (!in_array($languageCode, $siteaccessLanguages)) {
+        $siteaccessRepository = $this->configResolver->getParameter(
+            'repository',
+            null,
+            $siteaccess
+        );
+        $siteaccessRepository = $siteaccessRepository ?: $this->repositoryConfigurationProvider->pullDefaultRepository();
+        $currentRepository = $this->repositoryConfigurationProvider->getRepositoryConfig()['alias'];
+
+        if (!in_array($languageCode, $siteaccessLanguages) || $siteaccessRepository !== $currentRepository) {
```

Comment:
We're comparing string vs. string, please use strict check.

We've already discovered a similar issue in validation for choice lists that went undetected.

```suggestion
        if (!in_array($languageCode, $siteaccessLanguages, true) || $siteaccessRepository !== $currentRepository) {
```

Reply:
This fragment was here before.